src/pkg/notifier/formats/cloudevents.go

```diff
@@ -0,0 +1,110 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package formats
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+
+	"github.com/goharbor/harbor/src/controller/event"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/notifier/model"
+)
+
+const (
+	// CloudEventsFormat is the type for CloudEvents format.
+	CloudEventsFormat = "CloudEvents"
+
+	// extOperator is the key for the operator in the CloudEvents.
+	extOperator = "operator"
+)
+
+var (
+	// eventTypeMapping defines the mapping of harbor event type and CloudEvents type.
+	eventTypeMapping = map[string]string{
+		event.TopicDeleteArtifact:    eventType("artifact.deleted"),
+		event.TopicPullArtifact:      eventType("artifact.pulled"),
+		event.TopicPushArtifact:      eventType("artifact.pushed"),
+		event.TopicQuotaExceed:       eventType("quota.exceeded"),
+		event.TopicQuotaWarning:      eventType("quota.warned"),
+		event.TopicReplication:       eventType("replication.status.changed"),
+		event.TopicScanningFailed:    eventType("scan.failed"),
+		event.TopicScanningCompleted: eventType("scan.completed"),
+		event.TopicScanningStopped:   eventType("scan.stopped"),
+		event.TopicTagRetention:      eventType("tag_retention.finished"),
+	}
+)
+
+// eventType returns the constructed event type.
+func eventType(t string) string {
+	// defines the prefix for event type, organization name or FQDN or more extended possibility,
+	// use harbor by default.
+	prefix := "harbor"
```

Comment:
As your comments, should we create a configuration item with default value `harbor` ?

Reply:
Sounds good, but not in this pr scope, we can expose it to configuration if needed in the future.